src/app/engine/audio-backend.service.ts

```diff
@@ -110,11 +110,16 @@ export class AudioBackendService {
   }
 
   private loadBank(url: string) {
+    // on Safari we need to use callbacks using decodeAudioData method.
     this.http
       .get(url, { responseType: 'arraybuffer' })
-      .pipe(mergeMap((result) => this.context.decodeAudioData(result)))
+      .pipe(mergeMap((result) => new Promise((resolve) => {
+        this.context.decodeAudioData(result, (buffer) => {
+          resolve(buffer);
+        })
+      }).then((buffer) => buffer) ))
```

Comment:
It seems like this `then` is unneeded and can be omitted, as it returned the buffer unmodified?

Reply:
👍 true!